### ISSUE TITLE
furnace_recipe in Item Scripts seem a bit ... promiscuous?

### ISSUE THREAD

**Issue by rjohnson98126** [DenizenScript/Denizen-For-Bukkit#1764]
Hi

I dont know if that's a technical term.

but for Item Scripts, i cant be sure, but it seems the 'furnace_recipe' is bugged.   Of course, its probably more likely that I'm buggy, but i've been over and over this, and dont know what i could be doing wrong:

basically the scenario is this:  a custom item, say, "custom cobble" is supposed to smelt into "custom stone" and then be crafted into "custom brick"

The crafting part of this scenario works (Item Script recipe:)   that is, ONLY "custom stone" can be used to craft "custsom brick", as expected, and that regular stone will only craft into regular brick, as expected...

but the furnace_recipe part does NOT work as intended.    Yes, "custom cobble" will smelt into "custom stone", but the problem is that ANY cobblestone will ALWAYS smelt into "custom stone"

is this a bug?   i cant see how im doing this wrong...

https://one.denizenscript.com/haste/47730

**Comment by rjohnson98126** [comment]
here's the code in question, fi you don't feel like going to the denizenscript link

```
ArenaCobble:
  type: item
  material: i@cobblestone
  display name: Arena Cobble
  lore:
  - verified legit

ArenaStone:
  type: item
  material: i@stone
  display name: Arena Stone
  lore:
  - verified legit
  furnace_recipe: i@ArenaCobble

ArenaStoneBrick:
  type: item
  material: i@stone_brick[quantity=4]
  display name: Arena StoneBrick
  lore:
  - verified legit
  recipe: 
  - i@ArenaStone|i@ArenaStone|i@air
  - i@ArenaStone|i@ArenaStone|i@air
  - i@air|i@air|i@air
```

**Comment by mcmonkey4eva** [comment]
Furnace recipe docs:
```
    //   # You can specify a material that can be smelted into your item.
    //   # Note: This can overwrite existing furnace recipes.
    //   # Additional note: This does not support dynamic data currently, only specific material types can be used as a recipe.
    //   furnace_recipe: i@item
```
The behavior you see is known/expected in current setup.
This might be improvable through event hacking in Denizen internal systems - though you can easily replicate how that would work directly in script (until we add it internally) via listening to furnace-related events.

**Comment by rjohnson98126** [comment]
using the lastest dev push
`version: 1.0.2-SNAPSHOT (build 309), core version: 1.15 (Build 105)`

 the behavior is such that CustomBlock1 will smelt into the expected CustomBlock2 per the furnace_recipe

but vanillaBlock1 will not not smelt into anything.    fuel is consumed, but vanillaBlock1 is unchanged.

**Comment by mcmonkey4eva** [comment]
My notes on a possible fix, copy/pasted from my post on Discord:

we could probably actually Grade-A bullshit it into working with NMS
extend NMS class `RecipesFurnace` with an override to method `getResult(ItemStack)` (and any other methods as relevant) and use reflection hacks to override static final field `a` to direct to the custom handler
the custom class would need to gather a reference to the pre-existing value of `a` as its base impl (for when a script-given value is not present)
ideally we'd override `boolean a(ItemStack, ItemStack)` with a script validator, and upload recipes to the map (bypassing the Bukkit methods)
but that's private marked so I'm not sure if we can override that at all

**Comment by mcmonkey4eva** [comment]
Redirecting to #1851

**Comment by mcmonkey4eva** [comment]
Fixed by the resolution to #1851
